run_linux.go

```diff
@@ -716,11 +715,29 @@ func (b *Builder) generateHosts(rdir, hostname string, addHosts []string, chownO
 		}
 		hosts.Write([]byte(fmt.Sprintf("%s\t%s\n", values[1], values[0])))
 	}
-
 	if hostname != "" {
 		hosts.Write([]byte(fmt.Sprintf("127.0.0.1   %s\n", hostname)))
 		hosts.Write([]byte(fmt.Sprintf("::1         %s\n", hostname)))
 	}
+
+	// getLocalIP returns the non loopback local IP of the host
+	getLocalIP := func() string {
```

Comment:
@nalind @Luap99 do we have any better way then this to get the hosts ip?

Reply:
A host can validly have multiple network addresses, so this can return at most one address, or even no address.  What does `docker build` provide?

Reply:
Buildkit does:
```
#5 [2/3] RUN cat /etc/hosts
#5 0.346 127.0.0.1      localhost buildkitsandbox
#5 0.346 ::1    localhost ip6-localhost ip6-loopback
```

Reply:
For docker you have to use `--add-host=host.docker.internal:host-gateway` AFAIK and they use the default docker0 bridge ip address.
I think this solution is the best for buildah. For podman we use the CNI or slirp4netns gateway but this is also difficult to get.